### ISSUE TITLE
Spike: Scope of `COVID-19 Status`

### ISSUE THREAD

**Issue by brittneybrown** [revelry-foundation/nightingale#26]
The purpose of this Spike is to determine the scope of work to support users ability to manage their COVID-19 status in the app. I have a proposed solution which includes tracking testing as well as current COVID-19 statues. There is also a route to create an MVP version of this effort which only allows the user to enter/edit their COVID-19 status and not capture testing as a data point.

## Background
Per the track on the project [documentation](https://docs.google.com/document/d/1Ob3Z_RxdGALPhl3V0XqQhLLYLYKKmMRXHf3fGT640rE/edit), we are looking for users to maintain whether they are Negative or Positive. I would argue that asking a user to enter whether they have been tested and what those results are could also be a path worth investigating. 

According to the CDC, patients being tested for a current case of COVID-19 can [wait 1-2 days](https://www.cdc.gov/coronavirus/2019-ncov/testing/diagnostic-testing.html) for test results. [Antibody tests](https://www.cdc.gov/coronavirus/2019-ncov/testing/serology-overview.html) can produce results in mere minutes. Because of the varied nature of test result times, and the unknown nature of reinfections, I think we can assess and account for additional data points through the following HMW statement:

_How might a user who has been tested contribute meaningful data to the community as they wait for their results?_

## Proposed Headlines
User - COVID-19 Status - Have I Been Tested?
User - COVID-19 Status - I Enter My Test Result (including date test was administered)
User - COVID-19 Status - Remind Me to Enter My Test Result
User - COVID-19 Status - Add New Test Result (user starts flow again with new "test" entry)
Engineering - Alert for Positive COVID-19 Status to Corresponding Locations Histories
<details>
<summary>Testing Follow Up Flow</summary>

![Screen Shot 2020-05-27 at 8 22 32 AM](https://user-images.githubusercontent.com/48835404/83024176-97bf7e00-9ff3-11ea-9b47-5f93ead71299.png)

</details>

## MVP Headlines
User - COVID-19 Status - I am/am not positive (including date test was administered)
User - COVID-19 Status - Edit my status
Engineering - Alert for Positive COVID-19 Status to Corresponding Locations Histories

-----
#### PACT: Private Automated Contact Tracing 
https://pact.mit.edu/

#### GOP Plan to introduce data privacy around contact tracing privacy bill
https://www.commerce.senate.gov/2020/4/wicker-thune-moran-blackburn-announce-plans-to-introduce-data-privacy-bill
Writeup on bill where we can learn from some of the concerns being addressed
https://www.theverge.com/2020/5/1/21243977/gop-senators-contact-tracing-data-coronavirus-covid-19-privacy

#### Opinion piece on security/privacy in bluetooth tracing
https://www.wired.com/story/apple-google-contact-tracing-strengths-weaknesses/
Interesting note considering we are running our own servers
>_Even if the keys that the app uploads to a server can't identify someone, they could, for instance, be linked with the IP addresses of the phones that upload them. That would let whoever runs that server—most likely a government health care agency—identify the phones of people who report as positive, and thus their locations and identities._
_Apps can prevent anyone other than the server from eavesdropping on those IP addresses and identifying diagnosed users by using HTTPS encryption and also padding data they upload to obscure it, says Johns Hopkins' Green. But you still have to trust the app server itself not to collect and store identifying data from those uploads._

Idea around healthcare providers generating QR codes for patients to confirm their positive diagnosis
>_When doctors or nurses have determined that a patient is Covid-19-positive, they would tap a button to generate a confirmation code and give it to the patient, who then enters it into their contact-tracing app. A representative from Apple and Google's joint contact-tracing project said that their system similarly envisions that patients can't declare themselves infected without the help of a health care professional, who would likely confirm with a QR code._

**Comment by brittneybrown** [comment]
note to self: see if this has any HIPPA or other privacy complications

**Comment by brittneybrown** [comment]
#### PACT: Private Automated Contact Tracing 
https://pact.mit.edu/

#### GOP Plan to introduce data privacy around contact tracing privacy bill
https://www.commerce.senate.gov/2020/4/wicker-thune-moran-blackburn-announce-plans-to-introduce-data-privacy-bill
Writeup on bill where we can learn from some of the concerns being addressed
https://www.theverge.com/2020/5/1/21243977/gop-senators-contact-tracing-data-coronavirus-covid-19-privacy

#### Opinion piece on security/privacy in bluetooth tracing
https://www.wired.com/story/apple-google-contact-tracing-strengths-weaknesses/
Interesting note considering we are running our own servers
>_Even if the keys that the app uploads to a server can't identify someone, they could, for instance, be linked with the IP addresses of the phones that upload them. That would let whoever runs that server—most likely a government health care agency—identify the phones of people who report as positive, and thus their locations and identities._
_Apps can prevent anyone other than the server from eavesdropping on those IP addresses and identifying diagnosed users by using HTTPS encryption and also padding data they upload to obscure it, says Johns Hopkins' Green. But you still have to trust the app server itself not to collect and store identifying data from those uploads._

Idea around healthcare providers generating QR codes for patients to confirm their positive diagnosis
>_When doctors or nurses have determined that a patient is Covid-19-positive, they would tap a button to generate a confirmation code and give it to the patient, who then enters it into their contact-tracing app. A representative from Apple and Google's joint contact-tracing project said that their system similarly envisions that patients can't declare themselves infected without the help of a health care professional, who would likely confirm with a QR code._

**Comment by brittneybrown** [comment]
note to self: follow up with LM (@brittneybrown's lawyer friend) re this text
>Hi there! Haha, I love that attitude. It’s not my area of specialty but, if there’s tracing there needs to be a privacy policy, and the biggest things for that are CCPA (newish CA protection law) and GDPR (EU privacy rights). I’ll ping my coworker (b/c it is HIS specialty) and see if he can recommend something. One of my best friends at work (and a fellow mom) is big and privacy issues too, so I’ll see if she’s come across anything

**Comment by iamjoshfrank** [comment]
Solid thinking, @brittneybrown . I'm going to double-down on what I wrote in Slack now that I've had a chance to digest what you have pulled together here.

**tl;dr - we're definitely on the MVP (or "Core Loop") track right now**

---

[via Slack](https://revelry-public.slack.com/archives/C014AR56T1P/p1590606407357000?thread_ts=1590523037.158100&cid=C014AR56T1P)

1. We're not asking users to create accounts, b/c everything should be as private/anonymous as possible.
2. As of this moment, we're not planning to track at the centralized server level any kind of COVID-19 testing status. Only an individual user's local app should know if that individual is negative or positive.
3. When a user downloads and installs the app, we assume they are negative.
4. The two main reasons we care about negative/positive status right now is to:
  a. Help negative users compare their list of locations against a list of known hotspots, so that they know if they've inadvertently been exposed and should consider getting tested; and
  b. Provide users that become positive the opportunity to anonymously share their locations captured in the app starting [n] days prior to their test results (or maybe [n] days prior to when they indicate they first noticed symptoms).

So, in terms of "persisting data" re: negative/positive status and also re: test date vs. test results date, if we're persisting anything, it's only within the local app (b/c there are no stored user accounts remotely on the server to persist them to).

**Comment by iamjoshfrank** [comment]
For this week's iteration, we should keep status change as simple as possible and mostly use the "mark myself positive" action as a trigger for asking the user to anonymously share their location history (during the period they were likely contagious) for the benefit of the greater community.

This also means, _for this iteration,_ it probably makes the most sense to focus on **people taking a COVID-19 test** and do not build for the antibody use case.

Why? Because I think the antibody use case opens up a whole big can of worms we can't solve for in the next 2 days (how long ago might I have been infected? did I have symptoms or was I asymptomatic, and thus very difficult to know when I was infected? how long has the app tracked and stored locally my location data, because i might have been infected 2 months ago and only been using the app for 2 weeks? etc.)

**Comment by iamjoshfrank** [comment]
@brittneybrown - I think you should run with the MVP headlines. And I think the first two can be consolidated. I would like some clarification on what is intended with the third (Engineer alert).

> User - COVID-19 Status - I am/am not positive (including date test was administered)
> User - COVID-19 Status - Edit my status
> Engineering - Alert for Positive COVID-19 Status to Corresponding Locations Histories

See also my notes in your spreadsheet and my Comments above that might guide you in Issue writing for this Milestone.

**Comment by brittneybrown** [comment]
we have defined direction. closing
